app/src/main/java/com/github/se/stepquest/map/LocationViewModel.kt

```diff
@@ -0,0 +1,116 @@
+package com.github.se.stepquest.map
+
+import android.annotation.SuppressLint
+import android.location.Location
+import android.os.Looper
+import androidx.activity.ComponentActivity
+import androidx.compose.runtime.*
+import androidx.lifecycle.*
+import com.google.android.gms.location.*
+
+data class LocationDetails(val latitude: Double, val longitude: Double)
+
+class LocationViewModel : ViewModel() {
+  private var locationCallback: LocationCallback? = null
+  private var fusedLocationClient: FusedLocationProviderClient? = null
+  val locationRequired = MutableLiveData<Boolean>()
+  val currentLocation = MutableLiveData<LocationDetails>()
+  private val _allocations = MutableLiveData<List<LocationDetails>>()
+  val locationUpdated = MutableLiveData<Boolean>()
+
+  init {
+    locationRequired.postValue(false)
+    locationUpdated.postValue(false)
+  }
+
+  @SuppressLint("MissingPermission")
+  fun startLocationUpdates(context: ComponentActivity) {
+    if (fusedLocationClient == null) {
+      fusedLocationClient = LocationServices.getFusedLocationProviderClient(context)
+    }
+    locationCallback =
+        object : LocationCallback() {
+          override fun onLocationResult(p0: LocationResult) {
+            for (lo in p0.locations) {
+              // Update UI with location data
+              currentLocation.value = LocationDetails(lo.latitude, lo.longitude)
+              //              println("Location in view: ${currentLocation.value}")
```

Comment:
Are those prints were used for debug? Probably the pull request should be without them

Reply:
yes i delete it also